src/shaka_scheme/system/gc/GCNode.cpp

```diff
@@ -3,23 +3,39 @@
 //
 
 #include "shaka_scheme/system/gc/GCNode.hpp"
+#include "shaka_scheme/system/gc/GCData.hpp"
+
 
 namespace shaka {
 
     namespace gc {
+        GCNode::GCNode() :
+          gc_data(nullptr) {}
 
         GCNode::GCNode(GCData *data) {
             gc_data = data;
         }
 
         GCNode::~GCNode() {}
 
-        Data &GCNode::operator*() {
+        Data &GCNode::operator*() const {
             return this->gc_data->get_data();
         }
 
-        Data *GCNode::operator->() {
+        Data *GCNode::operator->() const {
             return this->gc_data->get_data_address();
         }
+
```

Comment:
One last thing -- @seanytak needs a `get()` lookalike as seen on `std::shared_ptr<Data>` because he'll need to be able to compare two pointers and see if they are pointing to the same piece of Data for implementing the equivalence predicates.

Reply:
Does this mean that the `GCData *` needs to be returned by this `get()` method?

Reply:
Never mind, I understand what is needed here.